torch/autograd/gradcheck.py

```diff
@@ -62,30 +63,48 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
     x_tensors = iter_tensors(target, True)
     j_tensors = iter_tensors(jacobian)
 
-    def compute_gradient(x, idx, is_mkldnn=False):
-
-        def fn_out():
-            if not is_mkldnn:
-                # x is a view into input and so this works
-                return fn(input).clone()
-            else:
-                # convert the dense tensor back to have mkldnn layout
-                return fn([x.to_mkldnn()])
-
-        orig = x[idx].item()
-        x[idx] = orig - eps
-        outa = fn_out()
-        x[idx] = orig + eps
-        outb = fn_out()
-        x[idx] = orig
-        r = (outb - outa) / (2 * eps)
-        return r.detach().reshape(-1)
+    def update_jacobians(x, idx, d, d_idx, is_mkldnn=False):
+
+        def compute_gradient(delta):
+            def fn_out():
+                if not is_mkldnn:
+                    # x is a view into input and so this works
+                    return fn(input).clone()
+                else:
+                    # convert the dense tensor back to have mkldnn layout
+                    return fn([x.to_mkldnn()])
+
+            orig = x[idx].item()
+            x[idx] = orig - delta
+            outa = fn_out()
+            x[idx] = orig + delta
+            outb = fn_out()
+            x[idx] = orig
+            r = (outb - outa) / (2 * eps)
+            return r.detach().reshape(-1)
+
+        ds_dx = compute_gradient(eps)
+        if x.is_complex():
```

Comment:
sooo... is this C -> C?

Reply:
> ```diff
> +    def __mul__(self, other):
> +        return self.__torch_function__(torch.mul, (Wrapper,), (self, other))
> ```

Then it fails with:

```
======================================================================
ERROR: test_gradcheck (__main__.TestGradCheckOverride)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_overrides.py", line 775, in test_gradcheck
    gradcheck(torch.add, (a, b), raise_exception=False)
  File "/home/chourdiaanjali/pytorch2/torch/autograd/gradcheck.py", line 461, in gradcheck
    outputs_to_check = [[torch._C._functions.UndefinedGrad()(o) for o in _differentiable_outputs(func(*tupled_inputs))]]
  File "/home/chourdiaanjali/pytorch2/torch/autograd/gradcheck.py", line 461, in <listcomp>
    outputs_to_check = [[torch._C._functions.UndefinedGrad()(o) for o in _differentiable_outputs(func(*tupled_inputs))]]
TypeError: argument 0 is not a Variable

----------------------------------------------------------------------
Ran 1 test in 0.097s
```

Reply:
C -> C, C -> R. also added a comment to make it more clear